src/Aspire.Hosting.Azure.AIFoundry/AzureAIFoundryResource.cs

```diff
@@ -0,0 +1,66 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Aspire.Hosting.ApplicationModel;
+
+namespace Aspire.Hosting.Azure;
+
+/// <summary>
+/// Represents an Azure AI Foundry resource.
+/// </summary>
+/// <param name="name">The name of the resource.</param>
+/// <param name="configureInfrastructure">Configures the underlying Azure resource using Azure.Provisioning.</param>
+public class AzureAIFoundryResource(string name, Action<AzureResourceInfrastructure> configureInfrastructure) :
+    AzureProvisioningResource(name, configureInfrastructure),
+    IResourceWithConnectionString,
+    IResourceWithEndpoints
+{
+    internal const string PrimaryEndpointName = "http";
+
+    private EndpointReference? _primaryEndpointReference;
+
+    private readonly List<AzureAIFoundryDeploymentResource> _deployments = [];
+
+    /// <summary>
+    /// Gets the "connectionString" output reference from the Azure AI Services resource.
+    /// </summary>
+    public BicepOutputReference AIFoundryApiEndpoint => new("aiFoundryApiEndpoint", this);
+
+    /// <summary>
+    /// Gets the connection string template for the manifest for the resource.
+    /// </summary>
+    public ReferenceExpression ConnectionStringExpression =>
+        ReferenceExpression.Create($"Endpoint={AIFoundryApiEndpoint}");
+
+    /// <summary>
+    /// Gets the list of deployment resources associated with the Azure AI Foundry.
+    /// </summary>
+    public IReadOnlyList<AzureAIFoundryDeploymentResource> Deployments => _deployments;
+
+    /// <summary>
+    /// Gets or sets a value indicating whether the resource is local.
+    /// </summary>
+    public bool IsLocal { get; set; }
```

Comment:
Now we have IsLocal, IsContainer, IsEmulator 😭

Reply:
We could use `IsEmulator` rather than `IsLocal`, since it's pretty analogous to an emulator in the way we use Foundry Local.